sdk/python/lib/pulumi/runtime/settings.py

```diff
@@ -232,8 +235,31 @@ def do_rpc_call():
 
     return SETTINGS.feature_support[feature]
 
+
 async def monitor_supports_secrets() -> bool:
     return await monitor_supports_feature("secrets")
 
+
 async def monitor_supports_resource_references() -> bool:
     return await monitor_supports_feature("resourceReferences")
+
+
+def reset_options(project: Optional[str] = None,
```

Comment:
Do we have an issue tracking the global state issues in the python SDK? 

We have an issue open for Node to track this. Might be worth creating a new issue for python or updating the node issue. https://github.com/pulumi/pulumi/issues/5449

Reply:
We do now! https://github.com/pulumi/pulumi/issues/6052